### DISCUSSION TITLE
Collecting Requirements/Ideas/Features for v1.9.0

### DISCUSSION THREAD

**Discussion by microwavables** [vmware-tanzu/velero#4529] [General discussions]
Hello Community!

Velero v1.8 is coming within the next few weeks, and before we start working on v1.9, we’d love to brainstorm with you on ideas and requirements for the feature list of this next release!

Please review what we have already laid out in [the v1.9 roadmap](https://github.com/vmware-tanzu/velero/wiki/1.9-Roadmap) and share your thoughts/ideas/requirements/feedback here so we can all work together on building out the next version of Velero.

**Comment by shawn-hurley** [comment]
Did we get credentials support per VSL in 1.8? I don't see this in 1.9 or did I miss it?

**Reply by reasonerjt** [reply]
@shawn-hurley it is not in v1.8 and it's not put into v1.9 roadmap.

see comment in https://github.com/vmware-tanzu/velero/issues/4493#issuecomment-1010285577 for why it's not done.
and there were also some update in https://github.com/vmware-tanzu/velero/issues/4115 where we decided to move it out of v1.8

**Reply by shawn-hurley** [reply]
Can we please put it on the v1.9 roadmap? is this a resources problem? What would we need to get on the v1.9 roadmap?

**Reply by dymurray** [reply]
@reasonerjt Just following up on this. Are we able to add this to the roadmap? My team would be happy to take on the implementation.

**Comment by shawn-hurley** [comment]
I would also like to discuss the use of parallelization in the backup code. 

From a high level the idea is that for a given item{which is defined as a kuberentes resource with a given namespace and name}, and any of it's additional items and actions, will be handled in a single thread, but that many of these items can be processed in parallel.


We do need to have a single thread writing to the TAR file for a given tar file which will then be uploaded.

**Reply by shawn-hurley** [reply]
Initial POC: #4444

**Comment by brito-rafa** [comment]
Here is my wishlist for Velero 1.9:

- Pre/Post Backup/Restore Plugin Hooks: https://github.com/vmware-tanzu/velero/pull/4083 
- Start deprecating Feature Flag EnableAPIGroupsVersions: https://github.com/vmware-tanzu/velero/issues/4547

**Reply by rPawel** [reply]
👍 for improved Hooks as per design spec!

**Comment by chrislinan** [comment]
Hi Community
I would like to know is there any plan to support HA ? Because currently velero only have one pod.
And Is there any solution that we can backup one cluster form another ? For example, I have ClusterA and ClusterB, I want to backup ClusterA use velero deployed in ClusterB.

**Reply by shawn-hurley** [reply]
This is a very interesting idea, the problem most likely will be getting the velero pod to talk to the cluster safely.

Maybe this is for my own edification, but how does this make velero HA in your opinion. if cluster A is down you can't back up or restore cluster A so I am trying to understand how velero running on cluster B talking to cluster A can help in this situation?

**Reply by chrislinan** [reply]
OK, I have two different ideas, first is velero support HA. And the second is we can backup clusterA from another cluster.

The second idea is base on the thought that if we can backup clusterA from another cluster, then we can use one velero backup many different clusters. Because velero backup has schedule and velero has nothing to do outside the schedule time. So can we use one velero backup clusterA on 10 a.m., backup clusterC on 11 a.m. , backup clusterD on 12.a.m and so on. Maybe that will reduce free time for velero and make velero busy, also we can save some resource on clusterA, clusterC and clusterD because we don't need to install velero on them.

**Comment by rPawel** [comment]
**Client-side encryption**, similar to https://github.com/vmware-tanzu/velero/issues/3218.
I want to encrypt the data on the cluster before I send it to s3 compatible storage. In our case, it is Backblaze B2.
I'd want to configure keys through Velero which would encrypt the data using GPG or any other secure encryption mechanism.
I am thinking of something similar to how GPG encryption is added to Duply: https://wiki.archlinux.org/title/Duply.
If one way or another the adversary gains access to the files they can not be decrypted - not your keys not your backup!
